lib/build.js

```diff
@@ -124,7 +124,59 @@ exports.filterIgnoreAndFiles = function(dir, ignore, files) {
           return;
         throw e;
       });
-    }));
+    }))
+    // remove empty folders
+    .then(function() {
+
+      //strip base dir from allFiles first to correctly compare it with the removeFiles
+      for (var i = allFiles.length - 1; i >= 0; i--) {
+        if(allFiles[i].indexOf(dir+'/' === 0)) allFiles[i] = allFiles[i].substr(dir.length+1);
+      }
+
+      // create object listing directories by file counts from the remaining files
+      var dirFileCount = {};
+      //var dirFileRemainCount = {};
+      allFiles.forEach(function(file) {
+        var dir = file.substr(0, file.lastIndexOf('/'));
+        dirFileCount[dir] = dirFileCount[dir] + 1 || 1;
+      });
+
+      // decrement file counts from removeFiles
+      removeFiles.forEach(function(file) {
+        var dir = file.substr(0, file.lastIndexOf('/'));
+        dirFileCount[dir] = dirFileCount[dir] - 1;
+      });
+
+      // now go through and populate removeDirs
+      var removeDirs = [];
+      Object.keys(dirFileCount).forEach(function(rdir) {
+        if (rdir!=='' && dirFileCount[rdir] === 0)
+          removeDirs.push(dir+'/'+rdir);
+      });
+
+      //add files from the removeFiles array that look like directories (do not have file extension)
+      //this is needed to pickup empty directories inside empty directories
+      removeFiles.forEach(function(file) {
+        var dotpos = file.lastIndexOf('.');
```

Comment:
yes that would be great.
i know what you mean now but i'm not sure how to tackle it.
i'm not in a rush or anything with this, so whenever you have time ;)